edgecenter/data_source_edgecenter_snapshot.go

```diff
@@ -0,0 +1,193 @@
+package edgecenter
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	edgecloudV2 "github.com/Edge-Center/edgecentercloud-go/v2"
+)
+
+func dataSourceSnapshot() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSnapshotRead,
+		Description: "Represent snapshot",
```

Comment:
A snapshot is a feature that allows you to capture the current state of the instance or volume at a specific point in time